### ISSUE TITLE
Slow initialisation before training and loss=nan with 3090

### ISSUE THREAD

**Issue by mocialov** [tensorflow/models#9525]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [x] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [x] I checked to make sure that this issue has not already been filed.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/research/object_detection/model_main_tf2.py

## 2. Describe the bug

I have two GPUs: 1) GeForce RTX 2080 (8GB) and 2) GeForce RTX 3090 (25GB) and I have been training object recognition models until today on the 1st GPU. Today I have tried switching to the 2nd GPU.

I am getting the following output after I run model_main_tf2.py:
`
2020-12-03 15:29:13.892211: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:39] Overriding allow_growth setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.
INFO:tensorflow:Using MirroredStrategy with devices ('/job:localhost/replica:0/task:0/device:GPU:0',)
I1203 15:29:13.897873 140206802110272 mirrored_strategy.py:500] Using MirroredStrategy with devices ('/job:localhost/replica:0/task:0/device:GPU:0',)
INFO:tensorflow:Maybe overwriting train_steps: 200000
I1203 15:29:13.900801 140206802110272 config_util.py:552] Maybe overwriting train_steps: 200000
INFO:tensorflow:Maybe overwriting use_bfloat16: False
I1203 15:29:13.900859 140206802110272 config_util.py:552] Maybe overwriting use_bfloat16: False
`

After around half an hour the training begins with loss=nan after every 100th epoch. The same exact source code, model, and config file, the model starts training after about 2-3 minutes with the loss not nan.

Why does it take such a long time to initialise before the training with another GPU and why I am getting nan loss on the same dataset and the same model, which show not nan loss during training on another GPU?

I am using Anaconda, with the environment being set up as follows (running from the models/research folder):
`
conda install cudatoolkit (installs v11.0.221)
pip install tensorflow-gpu=2.2 (v 2.2.0)
protoc object_detection/protos/*.proto --python_out=.
pip install .
pip install git+https://github.com/google-research/tf-slim
pip install pycocotools
export PYTHONPATH=$PYTHONPATH:../
`

## 3. Steps to reproduce

Get GeForce RTX 3090 (25GB)
pull https://github.com/tensorflow/models/tree/master/research/object_detection
run lines above

## 4. Expected behavior

Expect shorter waiting time before the training and not nan loss values

## 5. Additional context

Log here: https://pastebin.com/jHh7P7zV

## 6. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device name if the issue happens on a mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.2.0
- Python version: 3.8.5
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): 7.5.0
- CUDA/cuDNN version: v11.0.221
- GPU model and memory: GeForce RTX 3090 (25GB)

**Comment by pascal-hwky** [comment]
I'm having the same issue with an RTX 3080.

**Comment by ctc53** [comment]
Same issue here.

**Comment by mocialov** [comment]
The issue resolved once I upgraded all the libraries and dependenices

**Comment by tankienleong** [comment]
Hi @mocialov , I have the same issue after change RTX2060 to RTX3060 ti.

May I know how you upgrade all the libraries and dependenices?
